Snakefile.gsc.smk

```diff
@@ -33,29 +33,30 @@ if "tumour_lib" in config.keys():
 if "normal_lib" in config.keys():
     normal_lib = config["normal_lib"]
 
-# Find an output filename for the config we are generating
-if "create_config" in config.keys():
-    output_filename = config["create_config"]
-else:
-    output_filename = get_gsc_output_folder(
-        patient_id=config["id"],
-        tumour_lib=tumour_lib,
-        normal_lib=normal_lib,
-        pipeline_ver=pipeline_ver,
-        ploidetect_ver=ploidetect_ver,
-        project=config["project"] if "project" in config.keys() else None,
-    )
-    output_filename = os.path.join(output_filename, CONFIG_BASENAME)
-
-print(f"config: {output_filename}")
-
-rule gsc_config:
-    params:
-        ver=f'--pipeline-ver {pipeline_ver} --ploidetect-ver {ploidetect_ver}',
-        id=config["id"],
-        libs=f'-b {config["biopsy"]}' if "biopsy" in config.keys() else f'-t {config["tumour_lib"]} -n {config["normal_lib"]}',
-        project=f' --project {config["project"]}' if "project" in config.keys() else ""
+
+
+output_dir = get_gsc_output_folder(config["id"], tumour_lib, normal_lib, pipeline_ver, ploidetect_ver, project=config["project"] if "project" in config.keys() else None)
+temp_dir = get_ploidetect_temp_folder(config["id"], tumour_lib, normal_lib, pipeline_ver, ploidetect_ver, project=config["project"] if "project" in config.keys() else None)
+
+
+ruleorder: preseg > gsc_preseg
+
+
+rule all_gsc:
+    input:
+        expand("{output_dir}/cna.txt", output_dir=output_dir)
+
+rule gsc_preseg:
```

Comment:
What was the difference for preseg vs gsc_preseg, here?

Reply:
It appears that without gsc_preseg then the workflow interprets the temp_dir variable in preseg as defined in Snakefile, rather than as defined in Snakefile.gsc.smk. So we need to have a rule in this Snakefile (so that it only runs when running Snakefile.gsc.smk) with higher priority than preseg, which uses Snakefile.gsc.smk's defined temp_dir.